proposals/fleet-multiple-abm-vpp.md

```diff
@@ -0,0 +1,130 @@
+### Multiple ABM/VPP tokens
+
+- Figma: https://www.figma.com/design/j2M1heOh8eZD6LcUJks6HE/%239956-Add-multiple-Apple-Business-Manager-and-Volume-Purchasing-Program-connections?node-id=2-130&t=3vPRYgndVgzoD8e4-0
+- Issue: https://github.com/fleetdm/fleet/issues/9956
+
+- Constraints/bussines logic:
+    - VPP tokens can be assigned to: "All teams", "No team" or a team. Default is "All teams"
+    - ABM tokens can be assigned to: "No team" or a team. Default is "No team"
+    - Multiple ABM/VPP tokens per team.
+    - Same private key can be used for multiple tokens.
+
+### ABM: gameplan overview
+
+The ADE workflow is summarized here
+
+https://github.com/fleetdm/fleet/blob/0a2a48b6d89ab9e428509eb918724c60c245db60/tools/mdm/apple/glossary-and-protocols.md?plain=1#L216-L235
+
+Currently, Fleet manages a single token, and it does the process described there in a cron job every 30 seconds.
+
+To support multiple tokens, we need to:
+
+- On each cron run, do the process for each token that was uploaded.
+- For each host ingested via ABM, keep track of the token that was used to ingest it.
+- Every time we need to make an ABM call to assign an ADE profile, choose the right token to use for the host.
+
+### VPP: gameplan overview
+
+The VPP token is used to:
+
+- Get a list of available apps for a team
+- Assign an app to a host
+
+To support multiple tokens, we need to:
+
+- Add a form to retrieve all tokens available for a team or "no team" this is `(team OR no team) ∪ all teams`
+- To get a list of available apps: loop over each token and get the available apps
```

Comment:
To get a list of available apps for a given team (in order to add an app to that team), we'd list all apps with that team's token, then all apps with "all teams" token (if it exists) - as you describe in the line above - and then dedupe?

Reply:
That's correct, thanks! I added a note about deduping